### PR TITLE
fix partial refund

### PR DESCRIPTION
**Pull request by kknd113**
@sankalpk This needs go live asap, we are currently refunding too much if:
1. All shipments are shipped but there is one unshipped shipment.
2. Customer cancels the last unshipped shipment.

We used to have this bug where associated shipment cost was removed if last unshipped shipment was removed. So I put this patch in https://github.com/dotandbo/spree_retailops/commit/bbc993d3b0c1e2be2413e3b390570539da96ad34 to fix it.

Basically, we were refunding before the shipment cost was attached back.

### CODE REVIEW

**Comment by sankalpk** [comment]
:+1: this looks like it'll work.
